### ISSUE TITLE
Support selection pruning

### ISSUE THREAD

**Issue by Jefffrey** [datafusion-contrib/orc-rust#17]
Make use of file statistics, stripe statistics, column statistics, row group indexes, and bloom filters

Need way to expose this functionality so users (like datafusion) can utilize to efficiently query large ORC files, e.g. via predicate pushdown

**Comment by suxiaogang223** [comment]
I'd like to do some research for this PR.

**Comment by suxiaogang223** [comment]
It seems that there is no implementation of predicate pushdown in arrow-rs. I think the implementation in DataFusion can be used as a reference to determine what APIs need to be provided for ORC.
https://github.com/apache/datafusion/blob/main/datafusion/core/src/datasource/physical_plan/parquet/row_group_filter.rs

**Comment by Jefffrey** [comment]
> It seems that there is no implementation of predicate pushdown in arrow-rs. I think the implementation in DataFusion can be used as a reference to determine what APIs need to be provided for ORC. https://github.com/apache/datafusion/blob/main/datafusion/core/src/datasource/physical_plan/parquet/row_group_filter.rs

Yeah this issue is a port from when this repo was split from the datafusion code; I think it should still be applicable for having APIs in this crate that'll enable pushdown from the datafusion code

**Comment by suxiaogang223** [comment]
https://github.com/datafusion-contrib/orc-rust/issues/40
